src/app/api/advocates/database/AdvocateRepository.ts

```diff
@@ -0,0 +1,15 @@
+import db from "../../../../db";
+import { asc, count } from 'drizzle-orm';
+import { advocates } from "../../../../db/schema";
+
+export class AdvocateRepository {
+  async find(limit: number, offset: number) {
+    const data = await db.select().from(advocates)
+      .orderBy(asc(advocates.id))
+      .limit(limit) // the number of rows to return
+      .offset(offset); // the number of rows to skip
+
+    const resultCount = await db.select({ count: count() }).from(advocates);
```

Comment:
Might not be very efficient if attempting to count a bigger table. If given enough time will need to look at implementing something similar. 
https://stackoverflow.com/questions/7943233/fast-way-to-discover-the-row-count-of-a-table-in-postgresql